main/task.py

```diff
@@ -27,7 +30,35 @@ def send_mail_notification(subject, body, to=None, **kwargs):
       '#####################################################################',
       sender, to or sender, subject, body
     )
-  deferred.defer(mail.send_mail, sender, to or sender, subject, body, **kwargs)
+  if config.CONFIG_DB.mailgun_api_key and config.CONFIG_DB.mailgun_api_base_url:
+    deferred.defer(send_mailgun_message, sender, to or sender, subject, body)
+  else:
+    deferred.defer(mail.send_mail, sender, to or sender, subject, body, **kwargs)
+
+
+def send_mailgun_message(sender, to, subject, body):
+  mailgun_api_key = config.CONFIG_DB.mailgun_api_key
+  mailgun_api_base_url = config.CONFIG_DB.mailgun_api_base_url
+  url = '%s/messages' % mailgun_api_base_url
+  data = {'from': sender,
+    'to': to,
+    'subject': subject,
+    'text': body}
+  headers = {"Authorization":
+               "Basic %s" % b64encode("api:%s" % mailgun_api_key)}
+  try:
+    resp = urlfetch.fetch(
+      url=url,
+      payload=urlencode(data),
+      method=urlfetch.POST,
+      headers=headers,
+      validate_certificate=True,)
+    if resp.status_code != 200:
+      logging.error(
+        'Problem sending e-mail, status code: %s \n %s \n %s' % (resp.status_code, str(resp), str(data)))
```

Comment:
```python
    if resp.status_code != 200:
      logging.error('Problem sending email, status code: %s \n %s \n %s' % (
        resp.status_code, str(resp), str(data),
      ))
```

Notice `email` instead of `e-mail` ([trends](https://www.google.com/trends/explore?q=e-mail,email))